src/core/QueryManager.ts

```diff
@@ -147,6 +156,16 @@ export class QueryManager<TStore> {
     this.localState = localState || new LocalState({ cache });
     this.ssrMode = ssrMode;
     this.assumeImmutableResults = assumeImmutableResults;
+    this.documentTransform = documentTransform
+      ? defaultDocumentTransform
+          .concat(documentTransform)
+          // The custom document transform may add new fragment spreads or new
+          // field selections, so we want to give the cache a chance to run 
+          // again. For example, the InMemoryCache adds __typename to field 
+          // selections and fragments from the fragment registry.
+          .concat(defaultDocumentTransform)
+      : defaultDocumentTransform
```

Comment:
suggestion:

Have `DocumentTransform.concat(undefined)` do `return this` and here just
```suggestion
    // The custom document transform may add new fragment spreads or new
    // field selections, so we want to give the cache a chance to run 
    // again. For example, the InMemoryCache adds __typename to field 
    // selections and fragments from the fragment registry.
    this.documentTransform = defaultDocumentTransform.concat(documentTransform)
```

Reply:
I'm not sure I agree. While it would simplify this particular line of code, it might not be obvious what to expect in isolation.

Using arrays as example, if you `[1].concat(undefined)`, you don't get back `[1]`, but `[1, undefined]`. In other words, its actually returning a new array with the added value. If someone expects this transform to act similarly, it's unclear what an `undefined` transform would mean.